collect_app/src/main/java/org/odk/collect/android/formentry/DiskFormSaver.java

```diff
@@ -2,14 +2,15 @@
 
 import android.net.Uri;
 
+import org.odk.collect.android.logic.FormController;
 import org.odk.collect.android.tasks.SaveFormToDisk;
 import org.odk.collect.android.tasks.SaveToDiskResult;
 
 public class DiskFormSaver implements FormSaver {
 
     @Override
-    public SaveToDiskResult save(Uri instanceContentURI, boolean shouldFinalize, String updatedSaveName, boolean exitAfter, ProgressListener progressListener) {
-        SaveFormToDisk saveFormToDisk = new SaveFormToDisk(instanceContentURI, exitAfter, shouldFinalize, updatedSaveName);
+    public SaveToDiskResult save(FormController formController, boolean shouldFinalize, String updatedSaveName, boolean exitAfter, ProgressListener progressListener, Uri instanceContentURI) {
```

Comment:
This is a small thing but I'd keep the `Uri` first since that's what you're saving.